### PR TITLE
Clean up documentation (rustdoc)

### PR DESCRIPTION
**Pull request by jakelang**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
-/// use ewasm_api::{Hash, block_hash, finish_data};
-///
-/// #[cfg(target_arch = "wasm32")]
-/// #[no_mangle]
-/// pub extern "C" fn main() {
-///     let a: Hash = block_hash(1);
-///     finish_data(&a.bytes);
-/// }
-/// ```
+//! ewasm_api is a library used to interface with Ethereum's EEI in Ewasm, a set of enhancements to
```
Comment (axic) [inline src/lib.rs]: Make `Ewasm` a link to https://github.com/ewasm?

File: src/lib.rs [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+//! the Ethereum smart contract platform.
+//! ewasm_api exposes both a set of unsafe "native" functions representing the actual EEI
+//! functions, and a set of safe wrappers around them. It is recommended not to use the native
+//! functions as they do not perform bounds-checking.
+//!
+//! To use ewasm_api, simply include it as a dependency in your project.
+//!
```
Comment (axic) [inline src/lib.rs]: Should mention the existence of features (no_std, wee_alloc/qimalloc, debug, experimental).

File: src/bignum.rs [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
 pub mod native {
     extern "C" {
         pub fn bignum_mul256(a: *const u32, b: *const u32, ret: *const u32);
         pub fn bignum_umulmod256(a: *const u32, b: *const u32, modulo: *const u32, ret: *const u32);
     }
 }
 
-pub fn mul256(a: &Uint256, b: &Uint256) -> Uint256 {
+/// Unsigned 256-bit multiplication.
+pub fn umul256(a: &Uint256, b: &Uint256) -> Uint256 {
```
Comment (axic) [inline src/bignum.rs]: Well this isn't just a documentation update :)

Comment (axic) [reply src/bignum.rs]: The problem I have with the naming is that it goes against the "agreed" API. I'm not sure what's the best.

File: src/debug.rs [hunk 4]
```diff
@@ -1,8 +1,10 @@
 //! The native debug interface exposed to the ewasm contract. These functions are for testing
 //! purposes only. On a live VM, any bytecode trying to import these symbols will be rejected.
 
-use crate::types::*;
+use crate::types::StorageKey;
 
+/// The native interface for debugging functions. Use the wrappers unless you know what you're
```
Comment (axic) [inline src/debug.rs]: These are not exported so not sure they need to be documented. At least "Use the wrapper unless you know what you're doing" is pointless.

File: src/bignum.rs (hunk 3 above)
Comment (jakelang) [reply src/bignum.rs]: ah right i'll fix that

File: README.md [hunk 5]
```diff
@@ -12,9 +12,17 @@ Add the dependency, as usual:
 [dependencies]
 ewasm-api = "0.8"
 ```
+In your project, include the prelude:
 ```rust
-extern crate ewasm_api;
+use ewasm_api::prelude::*;
```
Comment (axic) [inline README.md]: Should mention this is an option, but isn't the only?

**Pull request merged**